cmd/k-license/k-license.go

```diff
@@ -0,0 +1,247 @@
+	"// Code generated by client-gen. DO NOT EDIT.",
+	"// Code generated by controller-gen. DO NOT EDIT.",
+	"// Code generated by counterfeiter. DO NOT EDIT.",
+	"// Code generated by deepcopy-gen. DO NOT EDIT.",
+	"// Code generated by informer-gen. DO NOT EDIT.",
+	"// Code generated by lister-gen. DO NOT EDIT.",
+	"// Code generated by protoc-gen-go. DO NOT EDIT.",
+}
+
+var codeFileExts = map[string]bool{
+	".go":    true,
+	".c":     true,
+	".h":     true,
+	".ipynb": true,
+	".py":    true,
+	".java":  true,
+	".cpp":   true,
+	".sh":    true,
+}
+
+var buildFileExts = map[string]bool{
+	"Makefile":   true,
+	"Dockerfile": true,
+}
+
+type Options struct {
+	templatesDir string
+	excludeDirs  []string
+	path         string
+	confirm      bool
+}
+type templateFileType struct {
+	fileExtension    string // store file extension strings like ".sh", "Makefile", etc.
+	templateFileName string // store template file names like "boilerplate.sh.txt", etc
+}
+
+var templateFileTypes = []templateFileType{
+	{".sh", "boilerplate.sh.txt"},
+	{"Makefile", "boilerplate.Makefile.txt"},
+	{"Dockerfile", "boilerplate.Dockerfile.txt"},
+	{".py", "boilerplate.py.txt"},
+	{".go", "boilerplate.go.txt"},
+}
+
+var opts = &Options{}
+
+func main() {
+	rootCmd := &cobra.Command{
+		Use:   "k-license",
+		Short: "Tool for Adding license Headers",
+	}
+	addCmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add Headers to files",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			return opts.Run()
+		},
+	}
+
+	addCmd.Flags().StringVar(&opts.templatesDir, "templates", "../../hack/boilerplate", "directory containing license templates")
+	addCmd.Flags().StringSliceVarP(&opts.excludeDirs, "exclude", "e", excludeDirsLocations, "comma-separated list of directories to exclude")
+	addCmd.Flags().StringVar(&opts.path, "path", ".", "Defaults to Current directory")
+	addCmd.Flags().BoolVar(&opts.confirm, "confirm", false, "")
+	rootCmd.AddCommand(addCmd)
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+
+}
+func (opts *Options) Run() error {
+	//excludeList := strings.Split(opts.excludeDirs, ",")
+	files := 0
+	err := filepath.WalkDir(opts.path, func(path string, info fs.DirEntry, err error) error {
+		if info.IsDir() && containsExcluded(opts.excludeDirs, info.Name()) {
+			return filepath.SkipDir
+		}
+		if !info.IsDir() && (isCodeFile(path) || isBuildFile(path)) && !isGenerateFile(path) {
+			hasLic, err := hasLicense(path)
+			if !hasLic {
+				currentYear := strconv.Itoa(time.Now().Year())
+				if opts.confirm {
+					err := addLicense(path, opts.templatesDir, currentYear)
+					if err != nil {
+						return err
+					}
+					fmt.Printf("Modified %s file\n", path)
+				}
+				files++
+			}
+			return err
+
+		}
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if opts.confirm {
+		fmt.Printf("Modified %v files\n", files)
+	} else {
+		fmt.Printf("Skipping modification. Dry running \t")
+		fmt.Printf("%v files will be modified\n", files)
```

Comment:
@mohitsharma-in, 

```
Skipping modification. Dry running 	1 files will be modified
```

In the above message, along with tracking the **count of files to be modified**, could we **also track & output their names (path)** as well? 

That would make above message more useful to understand where actual changes will be made.

Also, in case `--confirm` flag is **not set:**

let's print upfront (something like):

```
DRY RUN: No file changes will be made! To make file modifications, rerun the command with  "--confirm" flag
```

And then, if:
- (1) if **count(files) == 0,** just print something like - "All files have appropriate License Headers. No changes required."
- (2) if **count(files) >0**, let's print something like - "N files will be modified to add License Headers" and then list their names!

Reply:
ack